impeller/display_list/canvas.cc

```diff
@@ -1084,6 +1121,42 @@ void Canvas::SaveLayer(const Paint& paint,
         transform_stack_.back().transform.HasTranslation()
             ? Entity::RenderingMode::kSubpassPrependSnapshotTransform
             : Entity::RenderingMode::kSubpassAppendSnapshotTransform);
+
+    if (will_cache_backdrop_texture) {
+      FML_DCHECK(backdrop_data);
+      // If all filters on the shared backdrop layer are equal, process the
+      // layer once.
+      if (backdrop_data->all_filters_equal &&
+          !backdrop_data->shared_filter_snapshot.has_value()) {
+        // TODO(jonahwilliams): compute minimum input hint.
+        backdrop_data->shared_filter_snapshot =
+            backdrop_filter_contents->RenderToSnapshot(renderer_, {});
```

Comment:
I'm realizing that this can result in worse performance even after we've computed the minimum input hint.  Imagine 2 blurred rectangles whose corners touch.

Should we maybe just stick to sharing the input?  That will always be an improvement.

Reply:
sharing the input is a good GPU perfomrnce improvement but removing the filter sharing would discard the vast majority of the CPU overhead for many small filters. I'll show you some numbers tomorrow but even with extra blurring this is still generally faster.

We could also detect how much of the snapshot is used and add a heuristic based on covered %. i do think it would be a bit of an edge case, like someone that adds a single pixel blur to both the top left and bottom right corners, and then opts into (remember this is opt in) backdrop id.

Reply:
Maybe we should add a benchmark for this case then so we can quantify the worst case scenario.  Something like 2 squares on opposite sides of the screen animating sigma, group versus not group.  Maybe we could leave some room outside the union so the minimum input hint change can show up when we implement that.

Reply:
That sounds like a good idea! will do.